config/routes.rb

```diff
@@ -0,0 +1,31 @@
+Rails.application.routes.draw do
+  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+
+  root 'works#homepage'
+
+  resources :votes, only: [:index, :show, :create]
+
+  resources :works do
+    resources :votes, only: [:index, :create, :update]
+    member do
+      post 'upvote'
+    end
```

Comment:
interesting, but not quite what we want. You're correct that you want to nest something about `post` requests and upvoting within a work, because then we get to make a request that is inherently related to a work (and it'll definitely have a work ID associated with it when it's nested!)... You have something closer to what we expected below in this file, where you nested `resources :votes` into `resources :users`. In general, we know that vote are tied to both a user and a work, so therefore at the time of creating a vote (aka the "upvote"), we need to have a way to reference a work and reference a user. Using routes as a mechanism for having a way to reference a work is a good strategy.